ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_QMLSamples/mainSample.cpp

```diff
@@ -0,0 +1,680 @@
+// [Legal]
+// Copyright 2022 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "pch.hpp"
+
+// Qt headers
+#include <QApplication>
+#include <QDir>
+#include <QGuiApplication>
+#include <QJSEngine>
+#include <QQmlApplicationEngine>
+#include <QQmlContext>
+#include <QQmlEngine>
+#include <QQuickWindow>
+#include <QtGlobal>
+
+#ifdef QT_WEBVIEW_WEBENGINE_BACKEND
+#  include <QtWebEngineQuick>
+#endif // QT_WEBVIEW_WEBENGINE_BACKEND
+
+// Choose a manager based on the type of interface.
+// Add toolkit header for Cpp only
+#ifdef CPP_VIEWER
+#  include "SampleManager.h"
+#  include "ArcGISRuntimeEnvironment.h"
+#include "CategoryListModel.h"
+#include "DataItem.h"
+#include "DataItemListModel.h"
+#include "Esri/ArcGISRuntime/Toolkit/register.h"
+#include "Sample.h"
+#include "SampleCategory.h"
+#include "SampleListModel.h"
+#include "SampleSearchFilterModel.h"
+#include "SearchFilterCriteria.h"
+#include "SourceCode.h"
+#include "SourceCodeListModel.h"
+#include "SyntaxHighlighter/syntax_highlighter.h"
+#include "DrawOrderLayerListModel.h"
+#else
+// #  include "QmlSampleManager.h"
+#endif
+
+// #include "CategoryListModel.h"
+// #include "DataItem.h"
+// #include "DataItemListModel.h"
+#include "Esri/ArcGISRuntime/Toolkit/register.h"
+// #include "Sample.h"
+// #include "SampleCategory.h"
+// #include "SampleListModel.h"
+// #include "SampleSearchFilterModel.h"
```

Comment:
Were these commented out before? I recommend just removing them if they aren't needed.

Reply:
They were not. This is mainSample.cpp is uniquely used for QMLSamples only. Basically all these parts removed are to get the build to be successful but nothing else since the QML Samples App need not be fully functioning at this point. This was the simplest way forward.

Reply:
Oh right, I missed this was the QML sample app stuff. Do whatever works then :-)